java/src/main/native/src/cudf_jni_apis.hpp

```diff
@@ -58,6 +58,14 @@ jobject contiguous_table_from(JNIEnv *env, cudf::packed_columns &split, long row
 
 native_jobjectArray<jobject> contiguous_table_array(JNIEnv *env, jsize length);
 
+bool cache_group_by_result_jni(JNIEnv *env);
+
+void release_group_by_result_jni(JNIEnv *env);
+
+jobject group_by_result_from(JNIEnv *env, jobjectArray &groups);
+
+jobject group_by_result_from(JNIEnv *env, jobjectArray &groups, jlongArray &uniq_key_columns);
```

Comment:
I see no documentation for these functions and had difficulty in figuring out what they are doing. Please add doxygen explaining what they are.

Reply:
Your concern is right. But I did not find a way to return multiple types.

Reply:
Done